descriptor.md

```diff
@@ -0,0 +1,113 @@
+<!--[metadata]>
++++
+draft = true
++++
+<![end-metadata]-->
+
+# OpenContainers Content Descriptors
+
+OCI have several components that come to together to describe an image. 
+References between components form a [Merkle Directed Acyclic Graph (DAG)](https://en.wikipedia.org/wiki/Merkle_tree).
+The references in the _Merkle DAG_ are expressed through _Content Descriptors_.
+A _Content Descriptor_ or _Descriptor_, describes the disposition of targeted content.
+A _Descriptor_ includes the type of content, an independently-verifiable content identifier, known as a "digest" and the byte-size of the raw content.
+
+Descriptors SHOULD be embedded in other formats to securely reference external content.
+
+Descriptors MAY be used as an independent object to specify content.
+
+Implementations MUST verify fetched content against 
+
+## Properties
+
+- **`mediaType`** *string*
+
+  This REQUIRED property contains the MIME type of the referenced object.
+
+- **`digest`** *string*
+
+  This REQUIRED parameter is the digest of the content, meeting the requirements outlined in [Digests and Verification](#digests-and—verification).
+  Retrieved content SHOULD be verified against this digest before processing.
+
+- **`size`** *int*
+  This REQUIRED property specifies the size in bytes of the object.
+  This property exists so that a client will have an expected size for the content before validating.
+  If the length of the retrieved content does not match the specified length, the content should not be trusted.
+
+### Reserved
```

Comment:
On Thu, Jun 02, 2016 at 01:47:53PM -0700, Stephen Day wrote:
> > +### Reserved
> 
> I don't see how JSON-LD is related at all.

It allows you to unambiguously declare semantics at a per-property
level (e.g. “when I say ‘urls’, I mean
http://opencontainers.org/image/descriptor/urls” or “when I say
‘urls’, I mean http://example.com/overload/urls”)

> Do you have examples?

Plain vanilla:

  {
    "@context": "http://opencontainers.org/image/descriptor/1.0"
    "mediaType": "application/vnd.oci.image.manifest.v1+json",
    "size": 7682,
    "digest": "sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270"
  }

Vendor ‘urls’:

  {
    "@context": "http://opencontainers.org/image/descriptor/1.0",
    "mediaType": "application/vnd.oci.image.manifest.v1+json",
    "size": 7682,
    "digest": "sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270",
    "urls": {
      "@type": "http://example.com/overload/urls",
      "@value": ["a.example.com"]
    }
  }

v1.1 OCI descriptor with URLs:

  {
    "@context": "http://opencontainers.org/image/descriptor/1.1"
    "mediaType": "application/vnd.oci.image.manifest.v1+json",
    "size": 7682,
    "digest": "sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270"
    "urls": [
      "https://a.example.com/cas",
      "https://b.example.com/blobs"
    ]
  }